ipn/ipnlocal/local.go

```diff
@@ -612,12 +612,35 @@ func (b *LocalBackend) Start(opts ipn.Options) error {
 		b.logf("Start")
 	}
 
+	b.mu.Lock()
+
+	// The iOS client sends a "Start" whenever it UI screen comes
+	// up, just because it wants a netmap. That should be fixed,
+	// but meanwhile we can make Start cheaper here for such a
+	// case and not restart the world (which takes a few seconds).
+	// Instead, just send a notify with the state that iOS needs.
+	if b.state == ipn.Running &&
+		opts.Prefs == nil &&
+		opts.AuthKey == "" &&
+		b.stateKey == opts.StateKey &&
+		b.hostinfo != nil &&
+		b.hostinfo.FrontendLogID == opts.FrontendLogID {
+		b.logf("Start: already running; sending notify")
+		nm := b.netMap
```

Comment:
does this need cloning under the lock?

Reply:
there is no Clone on it.  I tried that first :)

But I think it's safe because netmaps are always created afresh in controlclient/direct.go and not mutated in place. The docs on it says you need to do your own clones if you need to mutate parts of it.

Reply:
Also: LocalBackend.setClientStatus already does the same as this code: it retains its provided new NetMap, then releases the lock, and sends a Notify with the same pointer value as retained.